### DISCUSSION TITLE
Cannot find text_plan_path file in steps outside of plan

### DISCUSSION THREAD

**Discussion by mennlo** [dflook/terraform-github-actions#174] [Q&A] [answered]
My use case is that I want to be able to run TerraCost on my TF plan. I am using Terraform Cloud with `agent` execution mode. This rules out the `json_plan_path`, so I was going to parse the TFC `run_id` out of the file provided by `text_plan_path`. I would then use the `json-output` feature of the TFC api to retrieve the json plan. However when I try to reference the file provided by that output, It is unreachable by subsequent steps.

During plan-terraform step `text_plan_path` is set to .dflook-terraform-github-actions/zgmnthba/plan.txt :
![image](https://user-images.githubusercontent.com/34140653/164737623-dfada7d5-ca28-44e9-bf5d-c75ee8a2ffa6.png)

Subsequent job cannot view that path:
![image](https://user-images.githubusercontent.com/34140653/164737799-6e1adb5d-47a3-4122-a0dd-17be56b53170.png)

**Comment by dflook** [comment]
Hello @mennlo, the path should be accessible from later steps in the _same_ job, It won't be available from subsequent jobs though.
You could try uploading it as an artefact in the plan job and then download it in the terracost job.

It sounds like the tfc `run_id` would be a useful output to add to the plan/apply actions. In fact I don't see a reason that the actions couldn't retrieve the json plan using the TFC API like you suggest, so that the `json_plan_path` always works.

**Reply by mennlo** [reply]
Thanks for the quick response, I am doing it within the same job, just a later step. It seems that is only available within the image / plan step. See below for the steps (plan and a debug step):

![image](https://user-images.githubusercontent.com/34140653/164754576-b2d64723-a88e-4a96-bccd-a493adfe2eb6.png)

See below for the output:

![image](https://user-images.githubusercontent.com/34140653/164754509-1bc7ff3b-70ab-4597-929d-01f4f3afddde.png)

**Reply by mennlo** [reply]
> In fact I don't see a reason that the actions couldn't retrieve the json plan using the TFC API like you suggest, so that the json_plan_path always works.

This would be a great enhancement.

**Reply by mennlo** [reply]
> Thanks for the quick response, I am doing it within the same job, just a later step. It seems that is only available within the image / plan step. See below for the steps (plan and a debug step):
> 
> ![image](https://user-images.githubusercontent.com/34140653/164754576-b2d64723-a88e-4a96-bccd-a493adfe2eb6.png)
> 
> See below for the output:
> 
> ![image](https://user-images.githubusercontent.com/34140653/164754509-1bc7ff3b-70ab-4597-929d-01f4f3afddde.png)

This seems to be solved by referencing it as `${{ github.workspace }}/${{ steps.plan-terraform.outputs.text_plan_path }}`. Unfortunately the url of the TFC run is not in that file. I wanted to use regex on that url to get the `run_id` and use that to retrieve the `json-output` from the TFC api.

**Comment by dflook** [comment]
dflook/terraform-plan@v1.25.0 creates the `json_plan_path` output for remote plans 🎉

**Comment by dflook** [comment, marked as answer]
The path outputs are relative to the actions workspace - usually that is the working directory. I can see in your output that the working-directory is being overwritten, which would be why you have to use an absolute path.
